From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alin Jerpelea <alin.jerpelea@sony.com>
Date: Thu, 22 Nov 2018 03:06:11 +0100
Subject: [PATCH] guard use of kernel sources

kernel is not compiled at build time and it should not be a dependency
Test: compiled OK without kernel sources

Change-Id: Ie8e892c5a7cca28cc58cbead88a9796ebc80a9f8
Signed-off-by: Alin Jerpelea <alin.jerpelea@sony.com>
---
 ipacm/src/Android.mk   | 2 ++
 ipanat/src/Android.mk  | 3 +++
 ipanat/test/Android.mk | 2 ++
 3 files changed, 7 insertions(+)

diff --git a/ipacm/src/Android.mk b/ipacm/src/Android.mk
index e7f5d21660fa3fd4888f6590135e3de6c84c1017..5a6fef8423e7e1f548e830aeb2bbc4bbd3aaa5e4 100644
--- a/ipacm/src/Android.mk
+++ b/ipacm/src/Android.mk
@@ -14,8 +14,10 @@ include $(CLEAR_VARS)
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/../src
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/../inc
 
+ifeq ($(TARGET_COMPILE_WITH_MSM_KERNEL),true)
 LOCAL_C_INCLUDES += $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
 LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
+endif
 
 LOCAL_CFLAGS += -DFEATURE_IPA_ANDROID
 LOCAL_CFLAGS += -DFEATURE_IPACM_HAL -Wall -Werror -Wno-error=macro-redefined -Wno-enum-compare
diff --git a/ipanat/src/Android.mk b/ipanat/src/Android.mk
index 4e955c07ad9dfd3929d3a167e711b3bb996cdb3b..c6caaeecb59696b64c4f16fd1c6bf0696538358b 100644
--- a/ipanat/src/Android.mk
+++ b/ipanat/src/Android.mk
@@ -10,8 +10,11 @@ LOCAL_PATH := $(call my-dir)
 include $(CLEAR_VARS)
 
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/../inc
+
+ifeq ($(TARGET_COMPILE_WITH_MSM_KERNEL),true)
 LOCAL_C_INCLUDES += $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
 LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
+endif
 
 LOCAL_SRC_FILES := ipa_nat_drv.c \
                    ipa_nat_drvi.c
diff --git a/ipanat/test/Android.mk b/ipanat/test/Android.mk
index b8ae6a4b1e2c19dd6e23a21c27e1f56f812c236c..e2b18886eca710115ebac76d351c0d9702946d67 100644
--- a/ipanat/test/Android.mk
+++ b/ipanat/test/Android.mk
@@ -10,8 +10,10 @@ include $(CLEAR_VARS)
 LOCAL_C_INCLUDES := $(LOCAL_PATH)/
 LOCAL_C_INCLUDES += $(LOCAL_PATH)/../../ipanat/inc
 
+ifeq ($(TARGET_COMPILE_WITH_MSM_KERNEL),true)
 LOCAL_C_INCLUDES += $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr/include
 LOCAL_ADDITIONAL_DEPENDENCIES := $(TARGET_OUT_INTERMEDIATES)/KERNEL_OBJ/usr
+endif
 
 LOCAL_MODULE := ipa_nat_test
 LOCAL_SRC_FILES := ipa_nat_test000.c \
